### PR TITLE
add replaceSubModel() to GUI

### PR DESCRIPTION
**Pull request by arun3688**
### Purpose

This PR add support to `replaceSubModel()` API in `OMEDIT`

### CODE REVIEW

**Review by adeas31 — CHANGES_REQUESTED** [review]
This is wrong. See how `mpElementPropertiesAction` is implemented.

**Comment by arun3688** [comment]
@adeas31 removed the `replaceSubModelSignal` and performing the `ReplaceSubModelDialog`  in the slot with this https://github.com/OpenModelica/OpenModelica/pull/9469/commits/c06ad6aa7e30f9c4edf38f7b8149b5c88590afc9

**Review by adeas31 — CHANGES_REQUESTED** [review]
Its still using the signal from GraphicsView and you are connecting/disconnecting from it in itemChange.

**Comment by arun3688** [comment]
@adeas31 Now all the actions and slots are moved to Element class https://github.com/OpenModelica/OpenModelica/pull/9469/commits/4434112221cc8208d4ac3422210a87cc6fa5b129

File: OMEdit/OMEditLIB/Element/Element.cpp [hunk 1]
```diff
@@ -3807,6 +3822,7 @@ void Element::showElementPropertiesDialog()
   }
 }
 
+
```
Comment (adeas31) [inline OMEdit/OMEditLIB/Element/Element.cpp:3825]: Remove this empty line.

**Review by adeas31 — CHANGES_REQUESTED** [review]
Looks better now. Please do the proposed changes and then hopefully we are good to go.

File: OMEdit/OMEditLIB/OMS/OMSProxy.cpp [hunk 2]
```diff
@@ -437,6 +437,26 @@ bool OMSProxy::addSubModel(QString cref, QString fmuPath)
...
+ * \param fmupath
+ * \param dryCount
+ * \param count
+ * \return
+ */
+bool OMSProxy::replaceSubModel(QString cref, QString fmuPath, bool dryCount, int *count)
+{
+  QString command = "oms_replaceSubModel";
+  QStringList args;
+  args << "\"" + cref + "\"" << fmuPath;
```
[the remark this answers is not in the document]
Comment (arun3688) [reply OMEdit/OMEditLIB/OMS/OMSProxy.cpp]: done

File: OMEdit/OMEditLIB/Element/Element.cpp [hunk 3]
```diff
@@ -2746,6 +2747,20 @@ void Element::createActions()
...
+  connect(mpReplaceSubModelAction, SIGNAL(triggered()), SLOT(showReplaceSubModelDialog()));
+}
+
+/*!
+ * \brief Element::showReplaceSubModelDialog
+ * Slot that opens up the ReplaceSubModelDialog Dialog.
+ */
+void Element::showReplaceSubModelDialog()
+{
+  ReplaceSubModelDialog *pReplaceFMUDialog = new ReplaceSubModelDialog(mpGraphicsView, this->getName());
```
[the remark this answers is not in the document]
Comment (arun3688) [reply OMEdit/OMEditLIB/Element/Element.cpp]: done with this https://github.com/OpenModelica/OpenModelica/pull/9469/commits/15e761d209907603f138ff663d36dd3600e7fb2b

File: OMEdit/OMEditLIB/Element/Element.cpp (hunk 1 above)
Comment (arun3688) [reply OMEdit/OMEditLIB/Element/Element.cpp]: done

**Review by adeas31 — APPROVED** [review]

**Pull request merged**